structuralcodes/sections/_section_results.py

```diff
@@ -0,0 +1,94 @@
+''' Results '''
+# import typing as t
+from dataclasses import dataclass
+
+
+@dataclass
+class GrossProperties:
+    """Simple dataclass for storing gross section properties."""
+
+    # section areas
+    area: float = 0
+    area_concrete: float = 0
+    area_reinforcement: float = 0
+
+    # axial rigidity
+    ea: float = 0
+
+    # section mass
+    mass: float = 0
+
+    # section perimeter
+    perimeter: float = 0
+
+    # first moments of area
+    sx: float = 0
+    sy: float = 0
+
+    # centroids
+    cx: float = 0
+    cy: float = 0
+
+    # second moments of area
+    i_yy: float = 0  # strong axis
+    i_zz: float = 0  # weak axis
+    i_yz: float = 0
+    # TODO: principal axes i_11 i_22 and angle
+
+    # section flexural rigidity
+    ei_xx: float = 0
+    ei_yy: float = 0
+
+    def __repr__(self, fmt: str = ".3f") -> str:
```

Comment:
This should be the `__str__` method since it prints an informal string representation of the object. I am not sure how common it is to have either `__str__` or `__repr__` with additional arguments though...

Reply:
I like the `__format__` way of approaching this :smiley:

Reply:
Included the `__format__` way of approaching it.